sound/snapcast/Makefile

```
@@ -0,0 +1,106 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=snapcast
+PKG_VERSION:=0.28.0
+PKG_RELEASE:=1
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=https://github.com/badaix/snapcast.git
+PKG_SOURCE_VERSION:=v$(PKG_VERSION)
+PKG_MIRROR_HASH=288325fc8b134502adb522711c18210794734675684cf5d8eba8a0a82778f6c9
+
+PKG_MAINTAINER:=Szabolcs Hubai <szab.hu@gmail.com>, David Andreoletti <david@andreoletti.net>
+PKG_LICENSE:=GPL-3.0-or-later
+PKG_LICENSE_FILES:=LICENSE
+PKG_CPE_ID:=cpe:/a:badaix:snapcast
+
+include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/cmake.mk
+
+CMAKE_OPTIONS += -DBUILD_TESTS=OFF
+
+define Package/snapcast/Default
+  SECTION:=sound
+  CATEGORY:=Sound
```

```suggestion
  CATEGORY:=Sound
  SUBMENU:=Snapcast
```

This might be the best choice for "grouping" 2 packages, without creating another dummy package.

What do you think?

---

> What do you think?


This would be my first package in `openwrt/packages` so my opinion is almost to none.
I'll just follow what you (and the other committers) suggests. :sweat_smile: 

As I can see this dummy `snapcast` package definition in `Makefile` it looks to me as a shortcut or a template to avoid repeating things for `snapclient` and `snapserver`.
I think this because of the `$(call Package/snapcast/*) calls.

I see something like that in `mpd`'s configuration: `Package/mpd/Default` contains all the reusable thing and there is to (variant) package of `mpd`.
And if I do `scripts/feeds install mpd` then all the dependencies and all `mpd` variant got installed into the build system.

It's OK to me to forget the grouping, but in my next PR (after I learn packaging) I plan to upgrade to latest Snapcast and plan to deal with OpenSSL support: I'd like to introduce `full` (with OpenSSL) and `mini` (without OpenSSL) (and of course some `snap{cast,server,client}-avahi-service` to deal with the Avahi dependency) variants which would be at least 2x2 = 4 different packages.
Would be fine to list all those `snap*` packages in the top level of "Sound" category?

As I wrote already, I am a package maintainer newbie, so I prefer what you suggests. :slightly_smiling_face:

---

> _provides_ the `mpd` package.

Ahh ohh ... so I won't have such a shortcut for Snapcast?
I mean: when I do `scripts/feeds install snapcast` I'd like to have all `snap*` installed in the build system. :thinking: 

I don't wanna do `scripts/feeds install snapserver` and `scripts/feeds install snapclient` (and `scripts/feeds install snapseserver-avahi-service`, once I got there) :confused:

---

Thank you very much! Looks like I really have to learn a lot!

---

> > Would be fine to list all those `snap*` packages in the top level of "Sound" category?
> 
> Yes for sure ;)

Went that way, dropped (sub) menu.
Thanks for the suggestion.